artifacts/4PS/Wait-DataUpgradeToFinish.ps1

```diff
@@ -20,44 +20,41 @@ function Wait-DataUpgradeToFinish {
         [string]$Tenant
     )
     PROCESS {
-        if (!$Tenant) {
-            $Tenant = 'default'
-        }
-
-        try {
-            Get-NAVDataUpgrade -ServerInstance $ServerInstance -Tenant $tenant -Progress
+        if (!$Tenant) { $Tenant = 'default' }
+        
+        try {      
+            do {
+                Start-Sleep -Seconds 1 | Out-Null
+                $status = Get-NAVDataUpgrade -ServerInstance $ServerInstance -Tenant $Tenant
+            } while ( "$($status.State)" -in @("InProgress") )

```

Comment:
The -Progress param should do the same, why did you change this?

Reply:
You're right, I just wanted to add the check below whether the progress equals 100% (which theoretically isn't really necessary) and adjusted the implementation to the one we use in the pipeline templates. 
I primarily just wanted to improve the output of the error messages, I can undo that if you want.

Reply:
I wasn't aware that we do something similar in pipelines. If that is the case, then I fully agree that doing it in the same way in both places totally makes sense 👍

Reply:
I went back to your solution as I like it more than the other one on the second guess (readability)
At a second glance, I went back to your solution as I prefer its readability. 👍